### ISSUE TITLE
Vod segments wrong order

### ISSUE THREAD

**Issue by LsHallo** [Brisppy/twitch-archiver#36]
**Describe the bug**
Every 10s the recorded vod (from live) jumps to another, completely different segment with a noticeable cut.
Completely unwatchable...
I'm already using a vpn to record to avoid ads. Not sure what might cause this.

**Supplied arguments**
--live-only -c [channel] -d "/output" -I "/config" -q 480p -w

**Error log**
<details>
<summary> Log </summary>
```
twitch-archiver  | 2024-06-22 12:36:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 12:37:38 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 12:39:00 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 12:39:04 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 12:43:21 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 13:03:22 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 13:11:40 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 13:13:51 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 13:18:15 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 14:03:34 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 14:21:00 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 14:24:37 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 15:06:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:08:16 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:17:46 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:23:19 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 15:24:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:25:14 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:27:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:27:36 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:29:14 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:29:54 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:30:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:32:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:32:45 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:33:24 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:35:26 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:37:49 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:41:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:48:21 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 15:51:14 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:52:16 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 15:53:48 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:56:40 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:56:58 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:57:36 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:58:38 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:59:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 15:59:56 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:00:18 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:04:24 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:06:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:20:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:31:45 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 16:52:27 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 17:11:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:11:44 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:14:38 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:14:47 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:16:07 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:16:25 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:18:13 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:24:48 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:42:54 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:47:26 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:50:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:51:55 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:52:46 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:53:54 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:54:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 17:59:46 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:00:07 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:01:47 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:03:04 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:04:38 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:06:46 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:07:36 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:08:50 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))
twitch-archiver  | 2024-06-22 18:09:08 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:11:00 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:16:31 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:16:58 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:18:44 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:20:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:27:45 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:32:26 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:34:47 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:37:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:45:12 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:48:25 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:48:34 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:53:46 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 18:57:17 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:02:55 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:07:34 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:10:48 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:12:26 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:12:35 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:12:44 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:14:34 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:16:58 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:18:09 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:18:14 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:18:18 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:19:06 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:19:28 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:19:45 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:27:05 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:36:38 [   ERROR] Exception encountered during GET request, retrying. Error: ('Connection aborted.', RemoteDisconnected('Remote end closed connection without response'))
twitch-archiver  | 2024-06-22 19:37:39 [    INFO] Merging VOD parts. This may take a while.
```
</details>

**Applicable files**
[VOD Link (first 90s)](https://ody.sh/oDI6e4U83M)

**Operating system**
Ubuntu 22.04 (docker container)

**Comment by Brisppy** [comment]
Thanks for opening an issue, this is likely due to your connection to Twitch being interrupted / unstable (which causes those RemoteDisconnected messages), combined with not having a VOD to verify the stream against. Can you run twitch-archiver with debug mode enabled `-D` and post the logs here? I'll see what I can do about narrowing the issue down from there.

**Comment by Brisppy** [comment]
Thanks for bringing this to my attention, a bug which affected streams without VODs caused segments to be overwritten in some cases.

Hopefully this is fixed with the current development build, I'll be doing some testing over the next few days to make sure.

**Comment by Brisppy** [comment]
This is now fixed with release [`v4.1.0`](https://github.com/Brisppy/twitch-archiver/releases/tag/v4.1.0) which will be fully released once I've finished testing.

**Comment by FlawTECH** [comment]
is this fixed on master branch ?
I cloned the projet 2 days ago but encountering this exact issue, on a stream which doesn't have VOD enabled.

I'm on commit 5abf7ee6441da811b48834278b3654b18be5a4f6, running on a Debian 12 system with ffmpeg 5.1.6-0+deb12u1

I can open another issue if that helps
